src/main/scala/firrtl/Compiler.scala

```diff
@@ -19,6 +19,22 @@ import firrtl.stage.Forms
 /** Container of all annotations for a Firrtl compiler */
 class AnnotationSeq private (private[firrtl] val underlying: List[Annotation]) {
   def toSeq: Seq[Annotation] = underlying.toSeq
+
+  /** Raw serialization of Firrtl compiler annotations for offline debug
+    *
+    * @see [[firrtl.options.phases.WriteOutputAnnotations]] for the canonical mechanism
+    *     of writing [[firrtl.options.StageOptions.annotationFileOut]]
+    */
+  def serialize: String =
+    JsonProtocol
+      .serializeTry(this)
+      .recoverWith {
+        case NonFatal(e) =>
+          val msg = s"Exception thrown during Annotation serialization:\n  " +
+            e.toString.replaceAll("\n", "\n  ")
+          Try(msg)
+      }
+      .get
```

Comment:
I don't think `AnnotationSeq` should depend on `JsonProtocol`. Instead I'd put this on `JsonProtocol` with a name like `serializeOrErrorMsg` or something.